java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java

```diff
@@ -0,0 +1,258 @@
+
+    PreStatementVisitor(SECheck check, CheckerContext context) {
+      super(context.getState());
+      this.check = check;
+      this.context = context;
+      this.constraintManager = context.getConstraintManager();
+    }
+
+    private void addZeroConstraint(boolean isZero) {
+      constraintManager.setValueFactory((id, node) -> new ZeroSymbolicValue(id, isZero));
+    }
+
+    private static boolean isZero(ProgramState programState, SymbolicValue symbolicValue) {
+      return programState.getConstraintWithStatus(symbolicValue, Status.ZERO) != null;
+    }
+
+    private void reportIssue(ExpressionTree expression, Tree.Kind kind) {
+      String operation = (kind == Tree.Kind.REMAINDER || kind == Tree.Kind.REMAINDER_ASSIGNMENT) ? "modulation" : "division";
+      String expressionName = expression.is(Tree.Kind.IDENTIFIER) ? ("'" + ((IdentifierTree) expression).name() + "'") : "this expression";
+      context.reportIssue(expression, check, "Make sure " + expressionName + " can't be zero before doing this " + operation + ".");
+
+      // interrupt exploration
+      programState = null;
+    }
+
+    @Override
+    public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+      List<SymbolicValue> symbolicValues = programState.peekValues(2);
+      boolean varIsZero = isZero(programState, symbolicValues.get(0));
+      boolean exprIsZero = isZero(programState, symbolicValues.get(1));
+
+      switch (tree.kind()) {
+        case MULTIPLY_ASSIGNMENT:
+          addZeroConstraint(exprIsZero);
+          break;
+        case MINUS_ASSIGNMENT:
+        case PLUS_ASSIGNMENT:
+          addZeroConstraint(varIsZero);
+          break;
+        case DIVIDE_ASSIGNMENT:
+        case REMAINDER_ASSIGNMENT:
+          if (exprIsZero) {
+            reportIssue(tree.expression(), tree.kind());
+          } else {
+            addZeroConstraint(varIsZero);
+          }
+          break;
+        default:
+          // do nothing
+          break;
+      }
+    }
+
+    @Override
+    public void visitBinaryExpression(BinaryExpressionTree tree) {
+      if (tree.is(
+        Tree.Kind.MULTIPLY, Tree.Kind.PLUS, Tree.Kind.MINUS, Tree.Kind.DIVIDE, Tree.Kind.REMAINDER,
+        Tree.Kind.GREATER_THAN, Tree.Kind.GREATER_THAN_OR_EQUAL_TO, Tree.Kind.LESS_THAN, Tree.Kind.LESS_THAN_OR_EQUAL_TO)) {
+        List<SymbolicValue> symbolicValues = programState.peekValues(2);
+        SymbolicValue rightOp = symbolicValues.get(0);
+        SymbolicValue leftOp = symbolicValues.get(1);
+        boolean rightOpIsZero = isZero(programState, rightOp);
+        boolean leftOpIsZero = isZero(programState, leftOp);
+        switch (tree.kind()) {
+          case MULTIPLY:
+            addZeroConstraint(rightOpIsZero || leftOpIsZero);
+            break;
+          case PLUS:
+          case MINUS:
+            addZeroConstraint(rightOpIsZero && leftOpIsZero);
+            break;
+          case DIVIDE:
+          case REMAINDER:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)